### PR TITLE
sql: split server from HTTPServer

### PR DESCRIPTION
**Pull request by tamird**
This separates SQL logic from the transport. It is necessary for experimentation with alternative transports.

### CODE REVIEW

File: sql/server.go [hunk 1]
```diff
@@ -139,81 +68,16 @@ func (s *Server) ServeHTTP(w http.ResponseWriter, r *http.Request) {
...
-	var errProto proto.Error
-	errProto.SetResponseGoError(err)
-	return driver.Result{Error: &errProto}
+	return reply, 0, nil
 }
 
 // exec executes the request. Any error encountered is returned; it is
 // the caller's responsibility to update the response.
-func (s *Server) exec(req driver.Request, planMaker *planner) driver.Response {
+func (s server) exec(req driver.Request, planMaker *planner) driver.Response {
```
Comment (vivekmenezes) [inline sql/server.go]: Let's call this execSQL

File: sql/server.go [hunk 2]
```diff
@@ -232,13 +96,13 @@ func (s *Server) exec(req driver.Request, planMaker *planner) driver.Response {
 	return resp
 }
 
-func (s *Server) execStmt(stmt parser.Statement, req driver.Request, planMaker *planner) (driver.Result, error) {
+func (s server) execStmt(stmt parser.Statement, req driver.Request, planMaker *planner) (driver.Result, error) {
```
Comment (vivekmenezes) [inline sql/server.go]: and this execSQLStmt

**Comment by vivekmenezes** [comment]
LGTM

File: sql/server.go (hunk 1 above)
Comment (tamird) [inline sql/server.go]: i've named this `execStmts` and left `execStmt` with the same name. I took the `req` argument out of both though, so they don't care about the wire type any more.

File: server/server.go [hunk 3]
```diff
@@ -141,7 +141,7 @@ func NewServer(ctx *Context, stopper *stop.Stopper) (*Server, error) {
 		}
 	}
 
-	s.sqlServer = sql.NewServer(&s.ctx.Context, s.db)
+	s.sqlServer = sql.NewServer(&s.ctx.Context, *s.db)
```
Comment (petermattis) [inline server/server.go]: `s/NewServer/NewHTTPServer/g`?

**Comment by petermattis** [comment]
LGTM

File: sql/server.go (hunk 2 above)
Comment (petermattis) [inline sql/server.go]: I have a mild preference for leaving `SQL` out of the name. Feels repetitive to have `sql.server.execSQL*`.

File: server/server.go (hunk 3 above)
Comment (tamird) [inline server/server.go]: done

**Pull request merged**